client/src/unifyfs-sysio.c

```diff
@@ -628,19 +605,8 @@ int unifyfs_fd_write(int fd, off_t pos, const void* buf, size_t count)
         return EOVERFLOW;
     }
 
-    /* get current log size before extending the log */
-    off_t logsize = unifyfs_fid_log_size(fid);
-
-    /* compute size log will be after we append data */
-    off_t newlogsize = logsize + count;
-
     /* finally write specified data to file */
-    int write_rc = unifyfs_fid_write(fid, pos, buf, count);
-    if (write_rc == 0) {
-        unifyfs_filemeta_t* meta = unifyfs_get_meta_from_fid(fid);
-        meta->needs_sync = 1;
-        meta->log_size = newlogsize;
-    }
+    int write_rc = unifyfs_fid_write(fid, pos, buf, count, bytes);
     return write_rc;
```

Comment:
So what happens if `write_rc == 0`, but `count != *bytes`?  Do we silently ignore that we didn't write enough bytes?  Would it make sense for `unifyfs_fid_write()` to instead return a `long`, equal to the number of bytes written or negative <ERRNO> on error?  That way we could encode all that information in a single value to pass it back up the stack.

Reply:
Here bytes is also an output param of fd_write, so it's the caller of fd_write that must check whether count == *bytes.  I figured we could end up with a short write either with error or without error so decided we should just pass that up the stack for someone else to figure out what to do with it.  This can return an error code indicating why it failed, so we end up with two output values: the return code for an error code and the number of bytes actually written.

Reply:
Sorry, I totally blanked and didn't see that `unifyfs_fd_write()` was returning `nwritten`.  Ignore me.